### PR TITLE
Accordion refactor

### PR DESCRIPTION
**Pull request by petertgiles**
Related to #6040 but does not close it.

## 👋 Introduction

This branch refactors the accordion component to make it possible to created headers needed for #6040 

## 🕵️ Details

- moves the experience accordions out of User Profile to the components directory
- Factors out the current accordion header as a StandardHeader component to maintain current appearance
  - renamed prop from `headerAs` to `headingAs` to reduce confusion :smile: 
- Separates the Accordion Header from the Accordion Trigger so that they can be used independently
- New Accordion Storybook story `WithEditLink` with what will eventually become the new ExperienceHeader

## 🧪 Testing

1. Rebuild app
2. Check that experience accordions look good
3. Check that non-experience accordions look good
  - both regular and simple styles
4. Check Chromatic diffs
5. Check the new Accordion/WithEditLink story

## 📸 Screenshot

image

### CODE REVIEW

File: packages/ui/src/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -48,6 +48,26 @@ const StyledHeader = React.forwardRef<
...
+}
+
+const Header = React.forwardRef<
+  React.ElementRef<typeof AccordionPrimitive.Header>,
+  AccordionHeaderProps
+>(({ headingAs = "h2", children, ...rest }, forwardedRef) => {
+  const Heading = headingAs;
+  return (
+    <StyledHeader asChild ref={forwardedRef} {...rest}>
+      <Heading data-h2-line-height="base(1)">{children}</Heading>
```
Comment (esizer) [inline packages/ui/src/components/Accordion/Accordion.tsx:66]: We will need to pass `headingAs` to this underlying component. Right now everything shows up at `h2` regardless of what is passed in.

File: packages/ui/src/components/Accordion/Accordion.stories.tsx [hunk 2]
```diff
@@ -35,30 +36,93 @@ const Template: ComponentStory<typeof Accordion.Root> = ({
...
+                data-h2-flex-direction="base(column)"
+                data-h2-gap="base(x.5 0)"
+              >
+                <span
+                  data-h2-display="base(block)"
+                  data-h2-font-size="base(h6, 1)"
+                >
+                  <span data-h2-font-weight="base(700)">Custom</span>
+                  {" Title"}
+                </span>
```
Comment (esizer) [inline packages/ui/src/components/Accordion/Accordion.stories.tsx:92]: I understand this is just a story for examples sake, but an [accordion trigger should always contain an element with `role=heading`](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/#:~:text=Each%20accordion%20header%20button%20is%20wrapped%20in%20an%20element%20with%20role%20heading%20that%20has%20a%20value%20set%20for%20aria%2Dlevel%20that%20is%20appropriate%20for%20the%20information%20architecture%20of%20the%20page.).

Comment (petertgiles) [reply packages/ui/src/components/Accordion/Accordion.stories.tsx:92]: Yeah, this is helpful.  It is more than an example - I plan to use this for the new experience accordions.

File: packages/ui/src/components/Accordion/Accordion.tsx (hunk 1 above)
Comment (petertgiles) [reply packages/ui/src/components/Accordion/Accordion.tsx:66]: Are you sure?  I popped an h3 into Storybook and it seemed to work OK.  Am I missing something?   :sweat_smile: 
image

Comment (esizer) [reply packages/ui/src/components/Accordion/Accordion.tsx:66]: Wow, I'm sorry. I completely missed line 64 there and was looking at the wrong story! 🤦‍♀️ 😛

File: packages/ui/src/components/Accordion/Accordion.stories.tsx (hunk 2 above)
Comment (petertgiles) [reply packages/ui/src/components/Accordion/Accordion.stories.tsx:93]: The radix header adds this role itself, doesn't it?
image

**Comment by petertgiles** [comment]
Alright, I feel pretty good about this Chromatic diff.  I think it's done!  Famous last words....  :sweat_smile:

File: packages/ui/src/components/Accordion/StandardHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import { Accordion } from "@gc-digital-talent/ui";
+import { AccordionHeaderProps } from "@gc-digital-talent/ui/src/components/Accordion/Accordion";
```
Comment (esizer) [inline packages/ui/src/components/Accordion/StandardHeader.tsx:2]: We will need to update this import path so the package isn't importing itself.

```tsx
import { Accordion } from "./Accordion";
```

Though, i'm wondering if it is worth just moving this to `Accordion.tsx` and exporting it as part of the `Accordion` object? So we could do: `<Accordion.StandardHeader>`. But that can be saved for #6412

**Review by esizer — APPROVED** [review]
LGTM once those imports get updated! 😁

File: packages/ui/src/components/Accordion/StandardHeader.tsx (hunk 3 above)
Comment (petertgiles) [reply packages/ui/src/components/Accordion/StandardHeader.tsx]: Good catch!  I've fixed the imports in [fix imports](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6384/commits/9c2a04f5e52f8a7437946ebc7664299e4f79ecca).

Yeah, I'll consider exporting the StandardHeader in the Accordion in the main PR, after seeing how the new ones integrates.   I'm not quite comfortable with the name though.  Is there a word for a header and trigger together?  I don't want to call it StandardHeaderAndTrigger.  :laughing:

Comment (esizer) [reply packages/ui/src/components/Accordion/StandardHeader.tsx]: heh, yeah... That is exactly what I was thinking. We could just make it obvious and say `Accordion.HeaderTrigger` 😅

**Pull request merged**